contract_manager/src/chains.ts

```diff
@@ -139,44 +203,78 @@ export class SuiChain extends Chain {
    * @param digest hex string of the 32 byte digest for the new package without the 0x prefix
    */
   generateGovernanceUpgradePayload(digest: string): Buffer {
-    let setFee = new SuiAuthorizeUpgradeContract("sui", digest).encode();
-    return this.wrapWithWormholeGovernancePayload(0, setFee);
+    const upgrade = new SuiAuthorizeUpgradeContract(
+      this.wormholeChainName,
+      digest
+    ).encode();
+    return this.wrapWithWormholeGovernancePayload(0, upgrade);
   }
 
   generateGovernanceSetFeePayload(fee: number, exponent: number): Buffer {
-    let setFee = new SetFee(
-      "sui", // should always be sui no matter devnet or testnet or mainnet
+    const setFee = new SetFee(
+      this.wormholeChainName,
       BigInt(fee),
       BigInt(exponent)
     ).encode();
     return this.wrapWithWormholeGovernancePayload(3, setFee);
   }
+
+  generateGovernanceSetDataSources(datasources: DataSource[]): Buffer {
+    const setDataSource = new SetDataSources(
+      this.wormholeChainName,
+      datasources
+    ).encode();
+    return this.wrapWithWormholeGovernancePayload(2, setDataSource);
+  }
 }
 
 export class EVMChain extends Chain {
   static type: string = "EVMChain";
 
-  constructor(id: string, public rpcUrl: string) {
-    super(id);
+  constructor(
+    id: string,
+    mainnet: boolean,
+    wormholeChainName: string,
+    private rpcUrl: string,
+    private networkId: number
+  ) {
+    super(id, mainnet, wormholeChainName);
   }
 
   static fromJson(parsed: any): EVMChain {
     if (parsed.type !== EVMChain.type) throw new Error("Invalid type");
-    return new EVMChain(parsed.id, parsed.rpcUrl);
+    return new EVMChain(
+      parsed.id,
+      parsed.mainnet,
+      parsed.wormholeChainName,
+      parsed.rpcUrl,
+      parsed.networkId
+    );
+  }
+
+  getRpcUrl(): string {
+    if (this.rpcUrl.includes("$INFURA_KEY") && !process.env.INFURA_KEY)
```

Comment:
nitpik: maybe it's better to do this check on the constructor

Reply:
I agree with ali on this, though also understand your point. Ideally we would have a way to only instantiate the chains the user requests, such that we can perform validity checks on instantiation. Not sure if that's easy to do though.

at the very least, this seems worth a code comment explaining why it's like this and what it would take to improve the situation.

Reply:
I decided to replace the current RPC endpoints with the public ones as the simplest solution and remove this logic😆